tests/test_upload.py

```diff
@@ -1006,199 +652,18 @@ def mock_api_call(self, operation_name, api_params):
-):
-    token = Token.objects.create(user=fakeuser)
-    permission, = Permission.objects.filter(codename="upload_symbols")
-    token.permissions.add(permission)
-    url = reverse("upload:upload_archive")
-
-    lookups = []
-
-    metadata_cache = {}
-
-    mock_bucket = gcsmock.MockBucket()
-
-    def mock_get_bucket(bucket_name):
-        return mock_bucket
-
-    gcsmock.get_bucket = mock_get_bucket
-
-    def mock_get_blob(key):
-        if key == "prefix/v0/flag/deadbeef/flag.jpeg":
-            return gcsmock.mock_blob_factory(key, size=69183)
-        if key == "prefix/v0/xpcshell.dbg/A7D6F1BB18CD4CB48/xpcshell.sym":
-            # Saying the size is 100 will cause the code to think the
-            # symbol file *is different* so it'll proceed to upload it.
-            size = 100
-            if lookups:
-                # If this is the second time, return the right size.
-                size = 501
-            lookups.append(size)
-            blob = gcsmock.mock_blob_factory(key, size=size)
-            if metadata_cache.get(key):
-                blob.metadata = metadata_cache[key]
-            return blob
-        raise NotImplementedError(key)
-
-    mock_bucket.get_blob = mock_get_blob
-
-    def mocked_create_blob(key):
-        blob = gcsmock.MockBlob(key)
-
-        def mock_upload_from_file(file):
-            pass
-
-        blob.upload_from_file = mock_upload_from_file
-        if key == "prefix/v0/xpcshell.dbg/A7D6F1BB18CD4CB48/xpcshell.sym":
-            pass
-        elif key == "prefix/v0/flag/deadbeef/flag.jpeg":
-            pass
-        else:
-            raise NotImplementedError(key)
-
-        return blob
-
-    mock_bucket.blob = mocked_create_blob
-
-    with open(ZIP_FILE, "rb") as f:
-        response = client.post(url, {"file.zip": f}, HTTP_AUTH_TOKEN=token.key)
-        assert response.status_code == 201
-
-        assert Upload.objects.all().count() == 1
-        assert FileUpload.objects.all().count() == 1
-
-    # Upload the same file again. This time some of the S3 HeadObject
-    # operations should benefit from a cache.
-    with open(ZIP_FILE, "rb") as f:
-        response = client.post(url, {"file.zip": f}, HTTP_AUTH_TOKEN=token.key)
-        assert response.status_code == 201
-
-        assert Upload.objects.all().count() == 2
-        assert FileUpload.objects.all().count() == 1
-        assert len(lookups) == 2
-
-    # Upload the same file again. This time some of the S3 HeadObject
-    # operations should benefit from a cache.
-    with open(ZIP_FILE, "rb") as f:
-        response = client.post(url, {"file.zip": f}, HTTP_AUTH_TOKEN=token.key)
-        assert response.status_code == 201
-
-        assert Upload.objects.all().count() == 3
-        assert FileUpload.objects.all().count() == 1
-        # This time it doesn't need to look up the size a third time
-        assert len(lookups) == 2
-
-
-@pytest.mark.django_db
-def test_upload_archive_key_lookup_cached_s3(
```

Comment:
The differ loses its place here, making it a little harder to review. What's going on:

* ``test_upload_archive_key_lookup_cached`` removed (GCS)
* ``test_upload_archive_key_lookup_cached_s3`` renamed to ``test_upload_archive_key_lookup_cached``
* ``test_upload_archive_key_lookup_cached_without_metadata`` removed (GCS)
* ``test_upload_archive_key_lookup_cached_without_metadata_s3`` renamed to ``test_upload_archive_key_lookup_cached_without_metadata``

The ``_without_metadata`` test is so similar that the diff gets confused, but it syncs again after these two tests.